README.md

```diff
@@ -59,12 +59,19 @@ Add +2 quantity of a single item to Cart and purchase in order to trigger an Err
 
 
 ## Deploy
-This run script deploys React + Flask. See the run script for individual gcloud commands you can use for any of the backends like /express, /ruby etc.
+This run script deploys React + Flask. See the run script for individual gcloud commands you can use for any of the backends like /express, /ruby etc. But if you're only deploying React, you need to use the full deploy.sh because it's running javascript build commands and sentry-cli.
```

Comment:
> But if you're only deploying React, you need to use the full deploy.sh because it's running javascript build commands and sentry-cli.

If you're only deploying react, do you need to comment out the flask deploy portion from `deploy.sh`? I feel this is getting overly complex for someone who doesn't know the ins and outs and have a suggestion.

How about separate `deploy_react.sh` and `deploy_flask.sh` scripts that are then stitched together in a single `deploy.sh`?

```
# deploy_react.sh

RELEASE=`./release.sh`
echo $RELEASE

SENTRY_ORG=testorg-az
SENTRY_PROJECT=application-monitoring-javascript
PREFIX=static/js

cd react
rm -rf build
npm install
npm run build

sentry-cli releases -o $SENTRY_ORG new -p $SENTRY_PROJECT $RELEASE
sentry-cli releases -o $SENTRY_ORG -p $SENTRY_PROJECT set-commits --auto $RELEASE
sentry-cli releases -o $SENTRY_ORG -p $SENTRY_PROJECT files $RELEASE upload-sourcemaps --url-prefix "~/static/js" --validate build/$PREFIX

# The release was set in the static prod build
gcloud app deploy --quiet
```

```
# deploy_flask.sh

cd ../flask && gcloud app deploy --quiet
```

```
# deploy.sh

source deploy_react.sh
source deploy_flask.sh
# possibly run through the whole list of other backends to deploy... or not
```

This way you can just run `deploy_react.sh` or `deploy_flask.sh` if you are only running one or the other. Other backends could also have their own deploy script i.e. `deploy_express.sh` or `deploy_springboot.sh`.

Reply:
I'm expecting that if anyone wants to deploy, they should understand and appreciate how it's done (this also helps make sure they don't overwrite the Master instance) and will know to check the deploy.sh for relevant commands.

I'm not looking to add several more deploy.sh scripts right now, each for a different platform, especially if all they're doing is running a gcloud command. 

For now I've created a react/README.md (as the other platforms have README's) and put Deploy instructions there. The express/README and springboot/README's also have Deploy sections with instructions. I also added a react/deploy.sh since it's more than 1 command. I guess now there's gcloud commands for some platform deployments, and a deploy.sh for the react one. That's inconsistent but I want to get to he pytests, and adding more shell scripts can also confuse people - it'd still be a readme but pointing to a shell script instead of a gcloud command. And when the shell script doesn't work, they''ll have to understand gcloud and app.yaml's anyways. They really shouldn't be deploying if they don't understand it.

I'd rather not test a new deploy.sh that calls deploy_react.sh deploy_flask.sh and have to re-test all. But this will be helpful as you may need to deploy 1 or both or multiple services all at once. Happy to attempt it after I get the pytests running. @cstavitsky

Reply:
@thinkocapo understood, that's fine, thanks for explaining your perspective